lib/declare_schema/field_declaration_dsl.rb

```diff
@@ -28,13 +28,12 @@ def optimistic_lock
       field(:lock_version, :integer, default: 1, null: false)
     end
 
-    def field(name, type, *args)
-      options = args.extract_options!
-      @model.declare_field(name, type, *(args + [@options.merge(options)]))
+    def field(name, type, *args, **options)
+      @model.declare_field(name, type, *[*args, @options.merge(options)])
     end
 
     def method_missing(name, *args)
-      field(name, args.first, *args[1..-1])
+      field(name, *args)
```

Comment:
I can't explain why the code was pulling the array apart and splatting it in two pieces when `*` already does that. The behavior is a bit different when there are no args given, but I don't think that matters since it's not valid.

Reply:
I can wait for this to go away with TECH-5112. Do we need a `**options` here has well since `method_missing` will be receiving both `arguments` and `keyword arguments`?

Reply:
If you just have `*`, that will absorb the options at the end as well.
```
def f(*args)
  args
end
=> :f
irb> f(1, 2, 3)
=> [1, 2, 3]
irb> f(1, 2, 3, a: :a, b: :b)
=> [1, 2, 3, {:a=>:a, :b=>:b}]
```